src/record.hpp

```diff
@@ -85,6 +86,23 @@ namespace geopm
                                  ///         parameter).
     };
 
+    /// @brief Format an event_e type as a string.
+    std::string event_name(int event_type);
```

Comment:
If these methods are going to be called a million times a second for populating the profile trace, should you be using string references wherever possible?

Reply:
Not for return values, that is resolved by return value optimization:

https://en.cppreference.com/w/cpp/language/copy_elision